packages/core/helper-plugin/src/utils/getYupInnerErrors.ts

```diff
@@ -0,0 +1,30 @@
+import type { TranslationMessage } from 'src/types/shared';
+import type { ValidationError } from 'yup';
+
+const extractValuesFromYupError = (
+  errorType: string | undefined,
+  errorParams: Record<string, any> | undefined
+) => {
+  if (!errorType || !errorParams) {
+    return {};
+  }
+
+  return {
+    [errorType]: errorParams[errorType],
+  };
+};
+
+const getYupInnerErrors = (error: ValidationError) =>
+  (error?.inner || []).reduce((acc, currentError) => {
+    if (currentError.path) {
+      acc[currentError.path.split('[').join('.').split(']').join('')] = {
+        id: currentError.message,
+        defaultMessage: currentError.message,
+        values: extractValuesFromYupError(currentError?.type, currentError?.params),
+      };
+    }
+
+    return acc;
+  }, {} as Record<string, TranslationMessage>);
```

Comment:
I think you can also use an index signature to avoid the type casting. Not really sure what's preferred / more readable though 🤔 

```suggestion
  (error?.inner || []).reduce<Record<string, TranslationMessage>>((acc, currentError) => {
    if (currentError.path) {
      acc[currentError.path.split('[').join('.').split(']').join('')] = {
        id: currentError.message,
        defaultMessage: currentError.message,
        values: extractValuesFromYupError(currentError?.type, currentError?.params),
      };
    }

    return acc;
  }, {});
```